xdsl/dialects/builtin.py

```diff
@@ -1691,6 +1697,24 @@ def iter_values(self) -> Iterator[float] | Iterator[int]:
     def get_values(self) -> tuple[int, ...] | tuple[float, ...]:
         return self.elt_type.unpack(self.data.data, len(self))
 
+    def get_int_values(self) -> Sequence[int]:
+        """
+        Return all the values of the elements in this DenseArray,
+        checking that the elements are integers.
+        """
+        el_type = self.get_element_type()
+        assert isinstance(el_type, IntegerType), el_type
+        return el_type.unpack(self.data.data, len(self))
+
+    def get_float_values(self) -> Sequence[float]:
+        """
+        Return all the values of the elements in this DenseArray,
+        checking that the elements are floats.
+        """
+        el_type = self.get_element_type()
+        assert isinstance(el_type, AnyFloat), el_type
+        return el_type.unpack(self.data.data, len(self))
+
```

Comment:
I would really like to avoid this. Surely we can print an array of floats and ints if we know the type without parsing the type? Seems like a similar mechanism to floats and ints of a known type in the assembly format.